frontend/catalyst/pennylane_extensions.py

```diff
@@ -168,54 +187,108 @@ class QJITDevice(qml.QubitDevice):
     pennylane_requires = "0.1.0"
     version = "0.0.1"
     author = ""
-    operations = [
-        "MidCircuitMeasure",
-        "Cond",
-        "WhileLoop",
-        "ForLoop",
-        "PauliX",
-        "PauliY",
-        "PauliZ",
-        "Hadamard",
-        "Identity",
-        "S",
-        "T",
-        "PhaseShift",
-        "RX",
-        "RY",
-        "RZ",
-        "CNOT",
-        "CY",
-        "CZ",
-        "SWAP",
-        "IsingXX",
-        "IsingYY",
-        "IsingXY",
-        "IsingZZ",
-        "ControlledPhaseShift",
-        "CRX",
-        "CRY",
-        "CRZ",
-        "CRot",
-        "CSWAP",
-        "MultiRZ",
-        "QubitUnitary",
-        "Adjoint",
-    ]
-    observables = [
-        "Identity",
-        "PauliX",
-        "PauliY",
-        "PauliZ",
-        "Hadamard",
-        "Hermitian",
-        "Hamiltonian",
-    ]
+
+    # These must be present even if empty.
+    operations = []
+    observables = []
+
+    @staticmethod
+    def _get_operations_available_in_QIR_abstract_machine():
```

Comment:
Any benefit over this being a field?

Reply:
https://github.com/PennyLaneAI/catalyst/pull/369/commits/6a43e045abffa04cd7d211510ae7d02a8bf4206b